src/wasm-ir-builder.h

```diff
@@ -264,6 +291,15 @@ class IRBuilder : public UnifiedExpressionVisitor<IRBuilder, Result<>> {
       return ScopeCtx(ElseScope{iff, originalLabel}, label);
     }
     static ScopeCtx makeLoop(Loop* loop) { return ScopeCtx(LoopScope{loop}); }
+    static ScopeCtx makeTry(Try* tryy, Name originalLabel = {}) {
+      return ScopeCtx(TryScope{tryy, originalLabel});
+    }
+    static ScopeCtx makeCatch(Try* tryy, Name originalLabel, Name label) {
```

Comment:
`originalLabel` is the name as it appears in the text input. `label` is a version of `originalLabel` that might have been modified to avoid having duplicate labels, which are allowed in the standard text format but not in Binaryen IR.

Reply:
Why do we need to keep `originalLabel` internally then? Is that because you try to roundtrip the labels as given? (If so, given that the code itself does not roundtrip anyway after being stored as a Binaryen IR, is it necessary?)

And why do `if` and `try` need only a `originalLabel` while `else` and `catch` needs both?

Reply:
We need to keep the scope's `originalLabel` around because it can be used by branch instructions inside the scope and we need to manage the mapping from `originalLabel` to internal `label`, including removing the mapping when the scope ends.

Reply:
Thanks. Why do if and try need only a originalLabel while else and catch needs both?

Reply:
Since the `label`s are only created lazily once we encounter a branch or delegate to a scope, it's not possible for there to be a nontrivial `label` at the beginning of an `if` or `try` scope. But at the beginning of an `else` or `catch` scope, there might have been a branch in the corresponding `if` or `try`, so we might have already generated a `label` that we need to propagate along to make sure branches in the `else` or `catch` go to the right place.